python/lsst/sims/photUtils/BandpassDict.py

```diff
@@ -0,0 +1,331 @@
+
+            if indices is not None:
+                fluxList = [numpy.NaN]*self._nBandpasses
+
+                fluxArray = dummySed.manyFluxCalc(self._phiArray, self._wavelenStep, observedBandpassInd=indices)
+                for i,ix in enumerate(indices):
+                    fluxList[ix] = fluxArray[i]
+            else:
+                fluxList = dummySed.manyFluxCalc(self._phiArray, self._wavelenStep)
+
+            return numpy.array(fluxList)
+
+        else:
+            return numpy.array([numpy.NaN]*self._nBandpasses)
+
+
+
+    def calcFluxListFromSedList(self, sedList, indices=None):
+        """
+        Return a 2-D array of fluxes from a SedList.
+        Each row will correspond to a different Sed, each column
+        will correspond to a different bandpass.
+
+        @param [in] sedList is a SedList containing the Seds
+        whose fluxes are desired.
+
+        @param [in] indices is an optional list of indices indicating which bandpasses to actually
+        calculate fluxes for.  Other fluxes will be listed as 'None' (i.e. this method will
+        return as many fluxes as were loaded with the loadBandpassesFromFiles methods; it will
+        just return nonsense for fluxes you did not actually ask for)
+
+        @param [out] output_list is a 2-D numpy array containing the fluxes
+        of each Sed (the rows) in each bandpass contained in this BandpassDict
+        (the columns)
+        """
+
+        one_at_a_time = False
+        if sedList.wavelenMatch is None:
+            one_at_a_time = True
+        elif len(sedList.wavelenMatch) != len(self._wavelen_match):
+            one_at_a_time = True
+        elif not numpy.allclose(sedList.wavelenMatch, self._wavelen_match, atol=0.0, rtol=1.0e-6):
+            one_at_a_time = True
+
+        output_list = []
+        if one_at_a_time:
+            for sed_obj in sedList:
+                sub_list = self.calcFluxListFromSed(sed_obj, indices=indices)
+                output_list.append(sub_list)
+        else:
+            if indices is not None:
+                for sed_obj in sedList:
+                    sub_list = numpy.array([numpy.NaN]*self._nBandpasses)
+                    if sed_obj.wavelen is not None:
+                        sed_obj.flambdaTofnu()
+                        flux_list = sed_obj.manyFluxCalc(self._phiArray, self._wavelenStep, observedBandpassInd=indices)
+                        for i,ix in enumerate(indices):
+                            sub_list[ix] = flux_list[i]
+                    output_list.append(sub_list)
+            else:
+                for sed_obj in sedList:
+                    if sed_obj.wavelen is None:
+                        sub_list = numpy.array([numpy.NaN]*self._nBandpasses)
+                    else:
+                        sed_obj.flambdaTofnu()
+                        sub_list = sed_obj.manyFluxCalc(self._phiArray, self._wavelenStep)
+                    output_list.append(sub_list)
```

Comment:
calcFluxListFromSed will perform the "is the wavelength grid of this sed the same as my _wavelen_match?" test.

if one_at_a_time = False, it means we already know that the wavelength grid is the same as the _wavelen_match, so we can avoid that test.

You're right that there is a lot of code duplication going on.  I'll create a private method that does all of the post-resampling calculations.